Add the request to the context for the serializer

pjrobertson opened the pull request:
This is useful if subclassing the views and using custom serializers.

Use case: User Signups is restricted to administrator users only. I set up a custom SignupSerializer that needs to validate the email address to make sure the domain matches the current user - the current user needs to be found from request.user

E.g.:
```python
def email_domain(email):
    # return email domain of 'email' - example.com

class EmailSignupSerializer(SignupSerializer):
    def validate_email(self, value):
        if email_domain(value) != email_domain(self.context['request'].user.email):
                raise serializers.ValidationError("You may only add users with from the same email domain as yours')
        return value
```

pjrobertson committed:
Add the request to the context for the serializer

This is useful if subclassing the views and using custom serializers.

Use case: I set up a custom SignupSerializer that needs to validate the email address to make sure the domain matches the current user - the current user needs to be found from request.user

merwok commented on authemail/views.py:
```diff
@@ -25,7 +25,7 @@ class Signup(APIView):
 
     def post(self, request, format=None):
-        serializer = self.serializer_class(data=request.data)
+        serializer = self.serializer_class(data=request.data, context={'request': request})
```
Would it be easier / better to call `context = self.get_serializer_context()` here?

(honest question, I use viewsets 99% of the time so don’t remember if view classes have the method)

pjrobertson replied:
Good spot, you're right!

In fact, `self.get_serializer(data=request.data)` should just be used instead. That defined in `generics.GenericAPIView` and calls through to `get_serializer_context()` so we don't have to.

pjrobertson replied:
Edit: nice idea, but `get_serializer` and `get_serializer_context` are only present on `generics.GenericAPIView` so it won't work here.

pjrobertson committed:
Raise exceptions in views instead of returning '400_BAD_REQUEST'

Allows django-rest to manage its own exceptions. Fixes #56

pjrobertson commented:
I just added one more commit that fixes #56

pjrobertson committed:
Improve password reset view:

1. Don't give away if a user account exists or not - for any valid email address, return a 201 - as if an email has been sent

pjrobertson committed:
Better validation on GET endpoints - make sure 'code' exists

pjrobertson committed:
Version bump

pjrobertson committed:
Also pass request to email templates, so that you can do URL resolving

pjrobertson committed:
Revert "Also pass request to email templates, so that you can do URL resolving"

This reverts commit a9b39f7c2e6bf7f68ed593ee920eb93ad395f479.

pjrobertson committed:
Fix Typo

pjrobertson committed:
Use X_FORWARDED_FOR for remote IP
